### DISCUSSION TITLE
How to access release tests from multistage yaml

### DISCUSSION THREAD

**Discussion by Jayz58** [rfennell/AzurePipelines#1289] [Q&A]
Hi, I can see my unit tests but is there a way to view my integration, functional, performance tests etc?

Also, is there a way to count the number of passes and fails?

**Comment by rfennell** [comment]
Any test that is published against a pipeline can be used. It is just a case of making sure you [publish your test results](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/publish-test-results?view=azure-devops&tabs=trx%2Cyaml). if this is done any integration, functional, performance tests are listed the sam as unit tests.

If you have Azure DevOps Manual Test Plans run against a pipeline these details are available using the following construct.

```
## Manual Test Plans
| Run ID | Name | State | Total Tests | Passed Tests |
| --- | --- | --- | --- | --- |
{{#forEach manualTests}}
| [{{this.id}}]({{this.webAccessUrl}}) | {{this.name}} | {{this.state}} | {{this.totalTests}} | {{this.passedTests}} |
{{/forEach}}
```
